code/modules/events/anomaly_bluespace.dm

```diff
@@ -43,9 +43,12 @@
 				var/y_distance = TO.y - FROM.y
 				var/x_distance = TO.x - FROM.x
 				for(var/atom/movable/A in range(12, FROM )) // iterate thru list of mobs in the area
-					if(istype(A, /obj/item/radio/beacon)) continue // don't teleport beacons because that's just insanely stupid
-					if(A.anchored) continue
-
+					if(istype(A, /obj/item/radio/beacon)) // don't teleport beacons because that's just insanely stupid
+						continue 
+					if(isobserver(A)) // don't teleport ghosts.
```

Comment:
Ghost has their move resist set to infinite. Instead of checking for observer, check for A.move_resist = INFINITY

Like this:
``` if(A.anchored || A.move_resist == INFINITY) ```

(Anchored and move_resist are bundled together since they're the same concept, except move_resist applies to moving creatures)